source/extensions/quic_listeners/quiche/platform/quiche_time_utils_impl.cc

```diff
@@ -0,0 +1,41 @@
+// NOLINT(namespace-envoy)
+
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include "extensions/quic_listeners/quiche/platform/quiche_time_utils_impl.h"
+
+namespace quiche {
+
+namespace {
+QuicheOptionalImpl<int64_t> QuicheUtcDateTimeToUnixSecondsInner(int year, int month, int day,
```

Comment:
Does this new code need tests?

Reply:
good point! Imported a quiche test: quiche_time_utils_test.cc